### ISSUE TITLE
The application force close while checking my news feed

### ISSUE THREAD

**Issue by bilelmoussaoui** [serge1peshcoff/photostream#8]
Hello, 
while checkingmy profile , the applications have been closed automaticlly ; and when i open it on terminal i've got this just before the applications close off
"Erreur de segmentation"
the whole log is over here : http://pastebin.com/w8BGfpvR

Thank you :)

**Comment by bilelmoussaoui** [comment]
PS : if you can add a gtk.spinner while the images are being downloaded (cache) its would be awesome !

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui there was a segfault while loading feed after specified interval, I know, the latest commit should fix it (I hope so).
About adding the spinner, that seems to be a good idea and I think I'll do it a bit later.

**Comment by bilelmoussaoui** [comment]
i tried the last built and stil lhave the same problem?

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui if you tried the last build, that's strange, that should've fixed it. This means there is another problem (or the same, but this didn't fix it). 
I don't have this problem, that's why it's strange.
Does it output only "segmentation fault" message?
I don't have a guess what can cause it, so I'm trying any guesses.

**Comment by bilelmoussaoui** [comment]
Yes i'm using the last build, btw the no user/tag has been fixed! the only thing i got is this 
[_LOG_LEVEL_FATAL 22:00:48.287351] [Gtk] _gtk_css_number_value_get: assertion 'number != NULL' failed
[_LOG_LEVEL_FATAL 22:00:48.287451] PhotoStream will not function properly.
Erreur de segmentation

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui and at what moment is this happening? can you provide full description of what you were doing?

**Comment by bilelmoussaoui** [comment]
i was just cheeking the news feeds; and waiting some pictures to load: sometimes the applications crush just after starting..i dont really know why if i can help with anything or test a new code i'm here !

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui I'm lost now.
I had some issues when loading pictures (some problems with SSL certificates), maybe that's the deal (again, this is only the guess).
Try opening the `src/Widgets/PostList.vala` and comment out everything inside `loadImages()` function (line 160-167 apparently) and rebuild (this will disable images loading at all, but at least we will know for sure if this is the same problem I've faced before or not).

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui btw, thanks a lot for helping me test it ;)

**Comment by bilelmoussaoui** [comment]
Ok i've commented everything inside the function,rebuild the application and when i open it i get this message error 
 JsonParse.vala:257: Something wrong with JSON parsing: <data>:1:1: Erreur d'analyse : unexpected character `<', expected value.
Trappe pour point d'arrêt et de trace

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui I've faced it before, this seems to be a problem with Instagram server, not my application (it sends not JSON, but some error like "service unavailable"). Try loading it a little bit later, hope it'll work.

**Comment by bilelmoussaoui** [comment]
Its working !! the application didnt crush at all!  And i think that i found an other bug? i open a new issue?
And can you tell me the the profile file? showing your own photos i'm going to try something

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui if it's working without images, that maybe the same issue I've faced before. still don't know how to fix it.
And if it is the same issue, for my case it didn't work for avatars but did work for images, try doing the same but commenting not full function but only `postBox.loadAvatar();` string, that should also work.
And yes, you are welcome to open a new issue.

And about profile, my Instagram username is `@pleasestaycalm`, or do you need something else?

**Comment by bilelmoussaoui** [comment]
Ops i was talking about the file name in src/widgets? and i will open it a little bit later.
i did rebuilt and i had this issue again, i guess the problem is from loadImage? 
I never said it, but sorry for my very bad english

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui user window (e.g. when you clicking "You" button in headerbar) is in `src/Windows/UserWindowBox.vala"`.
and if I understood right, you have segfaults after commenting out `loadAvatar` string and not commenting `loadImage`? if yes, that indeed means that the problem is in `loadImage` function (or in `loadAvatar` also, I don't know).

**Comment by bilelmoussaoui** [comment]
Yes as you said when i only comment loadAvatar , and thank you!

**Comment by serge1peshcoff** [comment]
Implemented feature in second post (now showing a spinner if image isn't loaded)

**Comment by bilelmoussaoui** [comment]
Hello, i think i found out the reason why the application force close! 
when you launch the application as a root from terminal the application is working so fine ! so maybe its a matter of permissions? 
And for the other bug : in a list of followers/users when you click on a user name i got this : 
[Gtk] Unable to show '@compass_student_network': Opération non prise en charge i guess you forgot to add an action when you click on it ?

**Comment by serge1peshcoff** [comment]
@bil-elmoussaoui maybe, maybe. if so, that is strange (it should close with error, not segfault).
About the second, I've faced it before, I know about it and I'll look at it a little bit later.
